### DISCUSSION TITLE
How to dynamically add children in TreeState

### DISCUSSION THREAD

**Discussion by xajhqffl** [adobe/react-spectrum#4473] [Q&A]
Hello,

We’ve implemented a tree view using the useTreeState hook. However, we want to support async loading of children nodes.

So after user click to expand tree node, we call api to get the children nodes.

but I don’t find a way to be able to modify the tree state by adding additional children.

Does anyone have any recommendations?

thank you!

**Comment by snowystinger** [comment]
I think you want `useTreeData` https://react-spectrum.adobe.com/react-stately/useTreeData.html
Which you can pass to things like `useTreeState`
